server/devdocs/bin/generate-components-usage-counts

```diff
@@ -0,0 +1,174 @@
+	var fileList = process.
+		argv.
+		splice( 2, process.argv.length ).
+		map( function ( fileWithPath ) {
+			return fileWithPath.replace( /^\.\//, '' );
+		} );
+
+	if ( fileList.length === 0 ) {
+		process.stderr.write( 'You must pass a list of files to process (try "make server/devdocs/usage-counts.js"' );
+		process.exit( 1 );
+	}
+
+	getModulesWithDependencies( root, fileList )
+		.then(  function ( modulesWithDependencies ) {
+			var stats = generateStats( modulesWithDependencies );
+			writeStats( stats, 'server/devdocs/usage-counts.js' );
+			process.exit( 0 );
+		} )
+		.catch( function ( error ) {
+			console.error( "An error occurred while processing the files: \n\t", error );
+			process.exit( 1 );
+		} );
+}
+
+/**
+ * Calculate the modules dependencies.
+ * Produces an object that has the following structure:
+ * {
+ *  'A': [dep1, dep2, dep3, ...],
+ *  'B': [dep1, dep2, dep3, ...]
+ * }
+ * @param {string} root BasePath to resolve the files' path
+ * @param {Array} fileList A list of file names
+ * @returns {Promise}
+ */
+
+function getModulesWithDependencies( root, fileList ) {
+	return new Promise( function ( resolve, reject ) {
+		var filesContent = {},
+		dependencies = {};
+
+		async.each(
+			fileList,
+			function ( fileWithPath, next ) {
+				fs.readFile(
+					fspath.join( root, fileWithPath ),
+					{ encoding: 'utf8' },
+					function ( err, data ) {
+						if ( err ) {
+							return next( err );
+						}
+						// get the dependencies
+						filesContent[ fileWithPath ] = data;
+						next( null );
+					}
+				);
+			},
+			function ( err ) {
+				if ( err ) {
+					return reject( err );
+				}
+				Object.keys( filesContent ).reduce(function ( target, module ) {
+					var deps = getDependencies( filesContent[ module ] );
+					if ( deps.length ) {
+						target[ module ] = deps;
+					}
+					return target;
+				} , dependencies );
+
+				return resolve( dependencies );
+			}
+		);
+	} );
+}
+
+/**
+ * Extract all required modules from a `code` string.
+ * Copyright (c) 2014-present, Facebook, Inc. See CREDITS.md#L1383
+ */
+function getDependencies( code ) {
+	var cache = Object.create(null);
+	var deps = [];
+
+	function addDependency( dep ) {
+		if ( !cache[ dep ] ) {
+			cache[ dep ] = true;
+			deps.push(dep);
+		}
+	}
+
+	code
+		.replace( replacePatterns.BLOCK_COMMENT_RE, '' )
+		.replace( replacePatterns.LINE_COMMENT_RE, '' )
+		.replace( replacePatterns.IMPORT_RE, function ( match, pre, quot, dep, post ) {
+			addDependency( dep );
+			return match;
+		} )
+		.replace( replacePatterns.EXPORT_RE, function ( match, pre, quot, dep, post ) {
+			addDependency( dep );
+			return match;
+		} )
+		.replace( replacePatterns.REQUIRE_RE, function ( match, pre, quot, dep, post ) {
+			addDependency( dep );
+		} );
+
+	return deps;
+}
+
+/**
+ * Given an object of modules with dependencies
+ * generate a stats object with the usage counts:
+ * {
+ *  'A': { count: 2 },
```

Comment:
Yes I can do that.
I prefer objects because they make it possible to add new data without breaking the previous version.
For example I was thinking that in a future version we could grab the module name / aliases from imports / requires too so that we can reference modules by name instead of path:

```javascript
// from
import Button from 'components/button';
// generate
{
  "components/button": {
    "alias": "Button",
    "count": 80
  }
}
// in the docs
count( 'Button' ) // or count( Button.displayName )
// instead of
count( 'components/button' )
```

That said, I am ok with just having an object with `moduleId` - `count` pairs.